tests/transient_key_context.rs

```diff
@@ -142,6 +164,33 @@ fn verify() {
         .expect("the signature should be valid");
 }
 
+#[test]
+fn verify_keypair() {
+    let pub_key = vec![
+        0xCD, 0x1A, 0xBA, 0xE5, 0xD7, 0x34, 0x34, 0x1A, 0xD3, 0x73, 0xBA, 0xE4, 0xF9, 0xEF, 0x46,
+        0xB1, 0xCF, 0x69, 0x9D, 0x40, 0x54, 0xC8, 0x59, 0xB9, 0xC0, 0xF0, 0xC8, 0x11, 0xCA, 0x4D,
+        0x7B, 0x1C, 0xB0, 0x3C, 0x66, 0xEA, 0x65, 0x51, 0x56, 0x63, 0x9B, 0x78, 0xC5, 0xDB, 0x2C,
+        0x2F, 0xEA, 0x42, 0x43, 0x0F, 0x41, 0x7A, 0xB3, 0xD4, 0xAE, 0xE5, 0xF6, 0x3B, 0x88, 0x1D,
+        0xD1, 0x06, 0xA3, 0xC6, 0x01, 0x05, 0xBC, 0x46, 0xBB, 0x18, 0xC7, 0xA7, 0x94, 0xA1, 0x7F,
+        0x50, 0x39, 0x24, 0x05, 0x55, 0x1F, 0x77, 0x28, 0x7E, 0x61, 0xB5, 0xF7, 0x84, 0x35, 0x4C,
+        0xD3, 0x51, 0x02, 0x1E, 0x18, 0x53, 0xB0, 0xCF, 0xD3, 0x47, 0x0D, 0x4C, 0xC9, 0xBD, 0x9E,
+        0x39, 0x83, 0x6B, 0x83, 0xC1, 0xBE, 0x6B, 0xB2, 0x00, 0xFE, 0xF5, 0x67, 0x86, 0x40, 0x6E,
+        0x8C, 0xD4, 0x5F, 0x73, 0xE4, 0xA9, 0xF5, 0x23,
+    ];
+
+    let priv_key = vec![
+        0xF6, 0x94, 0x95, 0x35, 0x2F, 0x2A, 0xB5, 0x8D, 0xB8, 0x9A, 0x0A, 0x6D, 0xDB, 0x06, 0x0C,
+        0xA0, 0xBA, 0xA5, 0xEC, 0x19, 0x0D, 0x1D, 0x61, 0xF0, 0xFA, 0xE3, 0x2C, 0xDF, 0xB7, 0x51,
+        0x6F, 0xC9, 0xE4, 0x96, 0x8B, 0x5C, 0x49, 0x4C, 0x05, 0x7F, 0x35, 0xDF, 0xE6, 0x91, 0x36,
+        0xFE, 0x35, 0x43, 0x4F, 0x0A, 0x3B, 0x89, 0x79, 0x55, 0x13, 0x47, 0xC4, 0x7A, 0x35, 0x7A,
+        0xBA, 0xD0, 0xAD, 0x0B,
+    ];
+
+    let mut ctx = create_ctx();
+    // exponent doesn't really matter here...
+    let _key_context = ctx.load_external_rsa(&priv_key, &pub_key, 0).unwrap();
```

Comment:
I guess `0` is allowed here because of (section 12.2.3.5 `TPMS_RSA_PARMS` of Part 2 "Structures" of the spec):

> A TPM compatible with this specification and supporting RSA shall support two primes and an exponentof zero. An exponent of zero indicates that the exponent is the default of (2^16)+1. Support for other values is  optional.  Use  of  other  exponents  in  duplicated  keys  is  not  recommended  because  the  resulting  keys would not be interoperable with other TPMs.

Shouldn't we handle that better here? As in, either:
* not having an exponent paramater at all
* indicating in the docs that 0 is a valid value and means that default will be selected (I think this is the minimum we should do)
* replace it with an `Option`? Or some other new type?

Reply:
Tbh I think we should document it and leave it there, we probs should support that being something other than 2^16 + 1 . I don't see `Option` as a better fit because `None` doesn't make more sense than `0` - and I think it's quite standard to regard `0` as that particular value by default. PSA Crypto generates all RSA keys with 2 ^ 16 + 1 public exponent, but I assume accepts any value for imports.

Reply:
An optional RsaExponent type wouldn't be that bad:

```rust
pub struct RsaExponent {
    value: u32,
}

impl Default for RsaExponent {
    fn default() -> RsaExponent{
        RsaExponent {value: 65537 }
    }
}

impl RsaExponent {
    pub fn new(value: u32) -> RsaExponent {
        RsaExponent { value }
    }

    pub fn value(&self) -> u32 {
        self.value
    }
}
```

And then in the api:
```rust
/// If no exponent is provided then it the default value will be used.
/// -->make fancy link to to RsaExponent default <--
    pub fn load_external_rsa(
        &mut self,
        key_prime: &[u8],
        public_modulus: &[u8],
        public_exponent: Option<RsaExponent>,
    ) -> Result<TpmsContext> {
.
.
.
        let rsa_parms = TpmsRsaParmsBuilder::new_unrestricted_signing_key(
            AsymSchemeUnion::RSASSA(HashingAlgorithm::Sha256),
            u16::try_from(public_modulus.len()).unwrap() * 8_u16,
            public_exponent.unwrap_or_default(),
        )
.
.
.
}
```

And then in the call:
```rust
let _key_context = ctx.load_external_rsa(&priv_key, &pub_key, None).unwrap();
```

I am not sure if I completely missed the point but to me this looks nice.

Reply:
Looks nice to me as well! I agree to both having an `Option` and passing `None` or just having the exponent so that callers directly pass `RsaExponent::default()`.

Reply:
I like the suggestion with the new type, but I still think I prefer not having an `Option` - `RsaExponent::default()` or `Default::default()` seems like a good fallback

Reply:
I have been convinced. New type and using default() is the best option.